### ISSUE TITLE
viewer Windows executable should include dll for portability

### ISSUE THREAD

**Issue by yanagiragi** [solvcon/solvcon#194]
Currently the viewer itself (whether user execute it via Visual Studio or not) is unable to perform as intended without set `$QT6_DIR\$VERSION\$PLATFORM\bin` to environment path first.

Not only it confuses to new user (As one needs to modify system path obviously is an unspoken rule), it also decreases the portability of the project.

After discussion, @tychuang1211 and I proposed three possible solutions: (other solutions are also welcomed!)
 
1. Add Qt binary paths to VS projects
    * pros: 
        * Easy to implement
    * cons: 
        * Only solves the case when the viewer is launched via Visual Studio
        * Poor portability 
 
1. Copy dll & necessary files into the folder of the executable when generate VS project
    * pros: 
        * Easy to implement
        * Good portability 
    * cons: 
        * Increase executable folder size
        * Developers had to aware of that the dlls are copied when CMake generates project.
    
1. Stay same for now. We'll built release build into single executable in future anyway.
    * pros: 
        * No extra work
    * cons: 
        * The unspoken rule remains
  
Let's discuss which solution fits us most in the moment.

**Comment by yungyuc** [comment]
@terrychan999 Do you think this is something we should do/consider for the [Windows nightly build](https://github.com/orgs/solvcon/projects/7/views/1)?

**Comment by terrychan999** [comment]
> @terrychan999 Do you think this is something we should do/consider for the [Windows nightly build](https://github.com/orgs/solvcon/projects/7/views/1)?

For the [Windows nightly build](https://github.com/orgs/solvcon/projects/7/views/1), I prefer "Approach 2". But the difference is that the DLL is copied after building the project because we use the windeployqt tool from Qt to copy the necessary DLLs.

**Comment by terrychan999** [comment]
I currently have a POC written in PowerShell that can make the viewer portable.
I have tested it on a virtual machine with a clean installation of Windows.

```ps1
$pyembed_url = "https://www.python.org/ftp/python/3.11.9/python-3.11.9-embed-amd64.zip"
$output = "python-3.11.9-embed-amd64.zip"
$destination = ".\mmdeploy\modmesh"
$get_pip_url = "https://bootstrap.pypa.io/get-pip.py"
$get_pip_script = "$destination\get-pip.py"
$pth_file = "$destination\python311._pth"


# Check if the file already exists
if (-Not (Test-Path -Path $output)) {
    # Download the Python Embedded ZIP file
    Invoke-WebRequest -Uri $pyembed_url -OutFile $output
}

# Clean build everytime
if (Test-Path -Path ".\build") {
    Remove-Item -Path ".\build" -Recurse -Force
}

# Create mmdeploy directory if it does not exist
if (-Not (Test-Path -Path $destination)) {
    New-Item -Path $destination -ItemType Directory
} else {
    Remove-Item -Path $destination -Recurse -Force
    New-Item -Path $destination -ItemType Directory
}


Expand-Archive -Path $output -DestinationPath $destination

# Enable Python Embedded  to install new packages
(Get-Content $pth_file) -replace '#import site', 'import site' | Set-Content $pth_file

# Download get-pip.py to the embedded Python directory
Invoke-WebRequest -Uri $get_pip_url -OutFile $get_pip_script

# Install pip in the embedded Python environment
& "$destination\python.exe" $get_pip_script

& "$destination\python.exe" -m pip install numpy matplotlib PySide6==6.7.0 shiboken6-generator==6.7.0

# Delete redundent Qt dlls from PySide6
# Get-ChildItem -Path "$destination\Lib\site-packages\PySide6\Qt*" -File | Remove-Item -Force


$pyside6_path = $(python -c "import sys, os, PySide6; sys.stdout.write(os.path.dirname(PySide6.__file__))") 
copy "$pyside6_path/pyside6.abi3.dll" $destination
$shiboken6_path = $(python -c "import sys, os, shiboken6; sys.stdout.write(os.path.dirname(shiboken6.__file__))")
copy "$shiboken6_path/shiboken6.abi3.dll" $destination


cmake -Dpybind11_DIR="$(pybind11-config --cmakedir)" -S . -B build
cmake --build build --config Release --target viewer

# Copy viewer.exe to mmdeploy directory
Copy-Item -Path ".\build\cpp\binary\viewer\Release\viewer.exe" -Destination $destination

# Run windeployqt on viewer.exe
windeployqt "$destination\viewer.exe"

# Copy modmesh directory to mmdeploy directory
Copy-Item -Path ".\modmesh" -Destination $destination -Recurse
# PUI is necessary
Copy-Item -Path ".\thirdparty" -Destination "$destination" -Recurse
```

**Comment by yungyuc** [comment]
It looks a good milestone.  @terrychan999 , let me assign this issue to you to track the progress.

**Comment by yungyuc** [comment]
@terrychan999 is working on this with PR #370.

**Comment by yungyuc** [comment]
Fixed by #370
